zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkConnection.java

```diff
@@ -36,16 +38,23 @@
 import org.apache.zookeeper.ZooKeeper.States;
 import org.apache.zookeeper.data.ACL;
 import org.apache.zookeeper.data.Stat;
-
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class ZkConnection implements IZkConnection {
-  private static final Logger LOG = Logger.getLogger(ZkConnection.class);
+  private static final Logger LOG = LoggerFactory.getLogger(ZkConnection.class);
 
   /** It is recommended to use quite large sessions timeouts for ZooKeeper. */
   private static final int DEFAULT_SESSION_TIMEOUT = 30000;
 
+  // A config to force disabling using ZK's paginated getChildren.
+  // By default the value is false.
+  private static final boolean GETCHILDREN_PAGINATION_DISABLED =
```

Comment:
Would this cause problem if default is enabled? Let's say server version is not supporting that. Then our getChildren will keep failing in looking up the method?

Reply:
It won't have a problem. If server does not support the feature, it will handle the `UnimplementedException` from server and fallback to the non-paginated API.